File: src/main/java/com/koddy/server/auth/infrastructure/oauth/zoom/ZoomOAuthProperties.java
package com.koddy.server.auth.infrastructure.oauth.zoom;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties("oauth2.zoom")
public record ZoomOAuthProperties(
        String grantType,
        String clientId,
        String clientSecret,
        String redirectUri,
        String authUrl,
        String tokenUrl,
        String userInfoUrl,
        Other other
) {
    public record Other(
            String createMeetingUrl,
            String deleteMeetingUrl
    ) {
    }
}

File: src/main/kotlin/com/koddy/server/auth/infrastructure/social/zoom/ZoomOAuthProperties.kt
package com.koddy.server.auth.infrastructure.social.zoom

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "oauth2.zoom")
data class ZoomOAuthProperties(
    val grantType: String,
    val clientId: String,
    val clientSecret: String,
    val redirectUri: String,
    val authUrl: String,
    val tokenUrl: String,
    val userInfoUrl: String,
    val other: Other,
)

data class Other(
    val createMeetingUrl: String,
    val deleteMeetingUrl: String,
)
